livebooks/elixir_ASCII_heart.livemd

```diff
@@ -0,0 +1,72 @@
+# Elixir goes ASCII Art ¯\_(ツ)_/¯
+
+## We💜Elixir, right?
+
+The aim is to create a heart-shaped pattern in your terminal that is made up of characters from the given string _Elixir_.
+
+The following code achieves that by using a combination of mathematical equations and string manipulation techniques.
+
+You'll find a step-by-step breakdown after the code. Let's go!
+
+```elixir
+defmodule Love do
+  @love "Elixir "
+
+  def character_at(@love, x, y) do
+    index = rem(x - y, String.length(@love))
+    char = String.slice(@love, index..index)
+
+    if (:math.pow(x * 0.05, 2) + :math.pow(y * 0.1, 2) - 1) ** 3 -
+         :math.pow(x * 0.05, 2) * :math.pow(y * 0.1, 3) <= 0 do
+      char
+    else
+      " "
+    end
+  end
+
+  def heart() do
+    -13..13
+    |> Enum.map(fn y ->
+      -30..30
+      |> Enum.map(fn x -> Love.character_at(@love, x, y) end)
+      |> Enum.join("")
+    end)
+    |> Enum.reverse()
+    |> Enum.join("\n")
+  end
+end
+
+IO.puts(Love.heart())
+```
+
+## Let's break it down
+
+The code is split into 5 parts:
+
+1. The definition of the module `WeLove` contains the global variable `love`, and the functions `character_at` and `heart`.
```

Comment:
Module attribute -> https://elixir-lang.org/getting-started/module-attributes.html

Reply:
Do you mean, I should call @love a module attribute?